src/agent/state/keypairs.rs

```diff
@@ -0,0 +1,282 @@
+        );
+    }
+
+    let primary_upload_route = {
+        let state = state.clone();
+        let rpc_url = primary_rpc_url.clone();
+        let min_balance = config.primary_min_keypair_balance_sol;
+        warp::path!("primary" / "load_keypair")
+            .and(warp::post())
+            .and(warp::body::content_length_limit(1024))
+            .and(warp::body::json())
+            .and(warp::path::end())
+            .and_then(move |kp: Vec<u8>| {
+                let state = state.clone();
+                let rpc_url = rpc_url.clone();
+                async move {
+                    let response = handle_new_keypair(
+                        state,
+                        Network::Primary,
+                        kp,
+                        min_balance,
+                        rpc_url,
+                        "primary",
+                    )
+                    .await;
+                    Result::<WithStatus<_>, Rejection>::Ok(response)
+                }
+            })
+    };
+
+    let secondary_upload_route = warp::path!("secondary" / "load_keypair")
+        .and(warp::post())
+        .and(warp::body::content_length_limit(1024))
+        .and(warp::body::json())
+        .and(warp::path::end())
+        .and_then(move |kp: Vec<u8>| {
+            let state = state.clone();
+            let rpc_url = secondary_rpc_url.clone();
+            async move {
+                if let Some(rpc_url) = rpc_url {
+                    let min_balance = config.secondary_min_keypair_balance_sol;
+                    let response = handle_new_keypair(
+                        state,
+                        Network::Secondary,
+                        kp,
+                        min_balance,
+                        rpc_url,
+                        "secondary",
+                    )
+                    .await;
+                    Result::<WithStatus<_>, Rejection>::Ok(response)
+                } else {
+                    Result::<WithStatus<_>, Rejection>::Ok(reply::with_status(
+                        "Secondary network is not active",
+                        StatusCode::SERVICE_UNAVAILABLE,
+                    ))
+                }
+            }
+        });
+
+    let http_api_jh = {
+        let (_, serve) = warp::serve(primary_upload_route.or(secondary_upload_route))
+            .bind_with_graceful_shutdown(config.bind_address, async {
+                let _ = crate::agent::EXIT.subscribe().changed().await;
+            });
+        tokio::spawn(serve)
+    };
+
+    // WARNING: All jobs spawned here must report their join handles in this vec
+    vec![http_api_jh]
+}
+
+/// Validate and apply a keypair to the specified mut reference,
+/// hiding errors in logs.
+///
+/// Returns the appropriate HTTP response depending on checks success.
+///
+/// NOTE(2023-03-22): Lifetime bounds are currently necessary
+/// because of https://github.com/rust-lang/rust/issues/63033
```

Comment:
I think this issue is now resolved, we might need to update rust tho. Maybe we can fix this in a future refactor PR.